Tropos/Models/DailyForecast.swift

```diff
@@ -0,0 +1,15 @@
+import Foundation
+
+@objc class DailyForecast: NSObject {
+    let date: NSDate
+    let conditionsDescription: String
+    let highTemperature: Temperature
+    let lowTemperature: Temperature
+    
+    init(JSON: Dictionary<String, AnyObject>) {
```

Comment:
Interesting. I didn't realize that you have to initialize all values on a class. That's pretty frustrating and I see how it would be a pain. Learned something new today.

What about something like this? Pretty much defining our own style of failable initializer

```diff
diff --git a/Tropos/Models/DailyForecast.swift b/Tropos/Models/DailyForecast.swift
index 0075d48..b9c9acd 100644
--- a/Tropos/Models/DailyForecast.swift
+++ b/Tropos/Models/DailyForecast.swift
@@ -1,15 +1,33 @@
 import Foundation
 
-@objc class DailyForecast: NSObject {
+@objc protocol Forecastable { }
+
+@objc class DailyForecast: Forecastable {
     let date: NSDate
     let conditionsDescription: String
     let highTemperature: Temperature
     let lowTemperature: Temperature
     
-    init(JSON: Dictionary<String, AnyObject>) {
-        self.date = NSDate(timeIntervalSince1970: JSON["time"] as! NSTimeInterval)
-        self.conditionsDescription = JSON["icon"] as! String
-        self.highTemperature = Temperature(fahrenheitValue: JSON["temperatureMax"] as! Int)
-        self.lowTemperature = Temperature(fahrenheitValue: JSON["temperatureMin"] as! Int)
+    init(time: NSTimeInterval, icon: String, temperatureMax: Int, temperatureMin: Int) {
+        self.date = NSDate(timeIntervalSince1970: time)
+        self.conditionsDescription = icon
+        self.highTemperature = Temperature(fahrenheitValue: temperatureMax)
+        self.lowTemperature = Temperature(fahrenheitValue: temperatureMin)
     }
+
+    static func forecastFromJSON(JSON: [String: AnyObject]) -> Forecastable {
+        if let time = JSON["time"] as? NSTimeInterval, icon = JSON["icon"] as? String, temperatureMax = JSON["temperatureMax"] as? Int, temperatureMin = JSON["temperatureMin"] as? Int {
+            return DailyForecast(time: time, icon: icon, temperatureMax: temperatureMax, temperatureMin: temperatureMin)
+        } else {
+            return NullDailyForecast()
+        }
+    }
+
+}
+
+@objc class NullDailyForecast: Forecastable {
+    let date: NSDate? = .None
+    let conditionsDescription: String? = .None
+    let highTemperature: Temperature? = .None
+    let lowTemperature: Temperature? = .None
 }
diff --git a/Tropos/Models/TRWeatherUpdate.m b/Tropos/Models/TRWeatherUpdate.m
index 56ce074..1cb24f6 100644
--- a/Tropos/Models/TRWeatherUpdate.m
+++ b/Tropos/Models/TRWeatherUpdate.m
@@ -53,7 +53,7 @@
     NSMutableArray *dailyForecasts = [NSMutableArray array];
 
     for (NSUInteger index = 1; index < 4; index++) {
-        DailyForecast *dailyForecast = [[DailyForecast alloc] initWithJSON:currentConditionsJSON[@"daily"][@"data"][index]];
+        DailyForecast *dailyForecast = [DailyForecast forecastFromJSON:currentConditionsJSON[@"daily"][@"data"][index]];
         [dailyForecasts addObject:dailyForecast];
     }
 
diff --git a/Tropos/ViewModels/TRDailyForecastViewModel.m b/Tropos/ViewModels/TRDailyForecastViewModel.m
index bb068a7..39f6ed5 100644
--- a/Tropos/ViewModels/TRDailyForecastViewModel.m
+++ b/Tropos/ViewModels/TRDailyForecastViewModel.m
@@ -30,19 +30,31 @@
     return [dateFormatter stringFromDate:self.dailyForecast.date];
 }
 
-- (UIImage *)conditionsImage
+- (nullable UIImage *)conditionsImage
 {
-    return [UIImage imageNamed:self.dailyForecast.conditionsDescription];
+    NSString *conditionsDescription = [NSString stringWithFormat:@"%@", self.dailyForecast.conditionsDescription];
+    return [UIImage imageNamed:conditionsDescription];
 }
 
-- (NSString *)highTemperature
+- (nullable NSString *)highTemperature
 {
-    return [[TRTemperatureFormatter new] stringFromTemperature:self.dailyForecast.highTemperature];
+    return [self stringFromTemperature:self.dailyForecast.highTemperature];
 }
 
-- (NSString *)lowTemperature
+- (nullable NSString *)lowTemperature
 {
-    return [[TRTemperatureFormatter new] stringFromTemperature:self.dailyForecast.lowTemperature];
+    return [self stringFromTemperature:self.dailyForecast.lowTemperature];
+}
+
+#pragma mark - Private
+
+- (nullable NSString *)stringFromTemperature:(nullable Temperature *)temperature
+{
+    if (temperature == nil) {
+        return @"";
+    } else {
+        return [[TRTemperatureFormatter new] stringFromTemperature:self.dailyForecast.lowTemperature];
+    }
 }
 
 @end
diff --git a/UnitTests/Tests/DailyForecastSpec.swift b/UnitTests/Tests/DailyForecastSpec.swift
index 3c531e5..6451ac7 100644
--- a/UnitTests/Tests/DailyForecastSpec.swift
+++ b/UnitTests/Tests/DailyForecastSpec.swift
@@ -7,12 +7,12 @@ class DailyForecastSpec: QuickSpec {
         describe("DailyForecast") {
             context("should correctly initialize set properties") {
                 let json: Dictionary<String, AnyObject> = ["time": Double(20.0), "icon": "text", "temperatureMax": 50, "temperatureMin": 10]
-                let dailyForecast = DailyForecast(JSON: json)
-                
-                expect(dailyForecast.date).to(equal(NSDate(timeIntervalSince1970: 20)))
-                expect(dailyForecast.conditionsDescription).to(equal("text"))
-                expect(dailyForecast.highTemperature.fahrenheitValue).to(equal(50))
-                expect(dailyForecast.lowTemperature.fahrenheitValue).to(equal(10))
+//                let dailyForecast = DailyForecast(JSON: json)
+//                
+//                expect(dailyForecast.date).to(equal(NSDate(timeIntervalSince1970: 20)))
+//                expect(dailyForecast.conditionsDescription).to(equal("text"))
+//                expect(dailyForecast.highTemperature.fahrenheitValue).to(equal(50))
+//                expect(dailyForecast.lowTemperature.fahrenheitValue).to(equal(10))
             }
         }
     }
```